apps/discovery_streams/test/integration/performance/cve_test.exs

```diff
@@ -0,0 +1,337 @@
+    bsm: File.read!(File.cwd!() <> "/test/integration/performance/bsm_message.json") |> Jason.decode!()
+  }
+
+  defmodule SetupConfig do
+    defstruct [
+      :messages,
+      prefetch_count: 0,
+      prefetch_bytes: 1_000_000,
+      max_bytes: 1_000_000,
+      max_wait_time: 10_000,
+      min_bytes: 0
+    ]
+  end
+
+  setup_all do
+    Logger.configure(level: :warn)
+    Agent.start(fn -> 0 end, name: :counter)
+
+    :ok
+  end
+
+  @tag timeout: :infinity
+  test "run performance test" do
+    map_messages = generate_messages(10_000, :map)
+    spat_messages = generate_messages(10_000, :spat)
+    bsm_messages = generate_messages(10_000, :bsm)
+
+    low_max_bytes = {"l", 1_000_000}
+    # mid_max_bytes = {"m", 10_000_000}
+    # high_max_bytes = {"h", 100_000_000}
+
+    low_max_wait_time = {"l", 1_000}
+    # mid_max_wait_time = {"m", 10_000}
+    # high_max_wait_time = {"h", 60_000}
+
+    low_min_bytes = {"l", 0}
+    # mid_min_bytes = {"m", 1_000}
+    # high_min_bytes = {"h", 1_000_000}
+
+    low_prefetch_count = {"l", 0}
+    # mid_prefetch_count = {"m", 100_000}
+    # high_prefetch_count = {"h", 1_000_000}
+
+    low_prefetch_bytes = {"l", 1_000_000}
+    # mid_prefetch_bytes = {"m", 10_000_000}
+    # high_prefetch_bytes = {"h", 100_000_000}
+
+    combos =
+      Combinatorics.product([
+        [{"spat", spat_messages}, {"bsm", bsm_messages}],
+        [low_max_bytes],
+        [low_max_wait_time],
+        [low_min_bytes],
+        [low_prefetch_count],
+        [low_prefetch_bytes]
+      ])
+
+    scenarios =
+      Enum.map(combos, fn l ->
+        {names, values} = Enum.unzip(l)
+
+        label = Enum.join(names, ".")
+
+        options =
+          Enum.zip([:messages, :max_bytes, :max_wait_time, :min_bytes, :prefetch_count, :prefetch_bytes], values)
+          |> Keyword.new()
+
+        {label, struct(SetupConfig, options)}
+      end)
+      |> Map.new()
+
+    Benchee.run(
+      %{
+        "kafka" => fn {dataset, expected_count, input_topic, socket} = _output_from_before_each ->
+
+          eventually(fn ->
+            current_total = AccumulatorTransportSocket.get_message_count(socket.transport_pid)
+
+            # IO.inspect("output is #{current_total} of #{expected_count}")
+
+            assert current_total >= expected_count
+          end, 100, 5000)
+
+          {dataset, input_topic, socket}
+        end
+      },
+      inputs: scenarios,
+      before_scenario: fn %SetupConfig{
+                            messages: messages,
+                            prefetch_count: prefetch_count,
+                            prefetch_bytes: prefetch_bytes,
+                            min_bytes: min_bytes,
+                            max_bytes: max_bytes,
+                            max_wait_time: max_wait_time
+                          } = _parameters_from_inputs ->
+        # existing_topic_config = Application.get_env(:discovery_streams, :topic_subscriber_config)
```

Comment:
whats all this for?

Reply:
this is usually where we'd tune the kafka consumer config, per the scenario. So we'd tweak the prefetch, etc. However, that's hardcoded inside of `Definition.Kafka` for the moment so we have it commented out